azure/services/virtualmachines/virtualmachines.go

```diff
@@ -165,6 +170,9 @@ func (s *Service) Reconcile(ctx context.Context) error {
 						Enabled: to.BoolPtr(true),
 					},
 				},
+				InstanceView: &compute.VirtualMachineInstanceView{
```

Comment:
do we want to support gen2 for vmss as well?

Reply:
yes we should

Reply:
is that even an option, can't find how to set it

Reply:
not sure but I feel like it should be, @devigned do you know?

Reply:
I think it should be. One sec... :eyes:

Reply:
I just created a VMSS with a the Ubuntu gen 2 image. There was nothing set in the VMSS PUT payload, so I'm thinking that VMSS gets that information from the image.

```
$ az vm image show -l westus2 -f UbuntuServer -p Canonical -s 19_10-daily-gen2 --version 19.10.202007100
{
  "automaticOsUpgradeProperties": {
    "automaticOsUpgradeSupported": false
  },
  "dataDiskImages": [],
  "disallowed": {
    "vmDiskType": "Unmanaged"
  },
  "hyperVGeneration": "V2",
  "id": "/Subscriptions/sub_id/Providers/Microsoft.Compute/Locations/westus2/Publishers/Canonical/ArtifactTypes/VMImage/Offers/UbuntuServer/Skus/19_10-daily-gen2/Versions/19.10.202007100",
  "location": "westus2",
  "name": "19.10.202007100",
  "osDiskImage": {
    "operatingSystem": "Linux",
    "sizeInBytes": 32213303808,
    "sizeInGb": 31
  },
  "plan": null,
  "tags": null
}
```

I'm questioning whether or not we need to update the `InstanceView`. I don't think that should be needed b/c the Compute resource provider should be responsible for determining the Hyper V generation based on the image used.